src/index.js

```diff
@@ -19,50 +19,34 @@ import { deviceTypes, MOBILE_DEVICE, CHROME_BROWSER, FIREFOX_BROWSER, IE_BROWSER
 import { deviceToPropMap, evaluate, Singleton } from './utils/index';
 
 export default function reactDeviceDetection (...config) {
-    return WrappedComponent => class extends React.Component {
+    return (WrappedComponent) => {
+        const uaParser = Singleton.instance(UAParser);
+        const deviceData = uaParser.getDevice();
+        const browserData = uaParser.getBrowser();
 
-        constructor () {
-            super();
-
-            this.state = {
-                isChrome: false,
-                isIE: false,
-                isMobile: false,
-                isFirefox: false,
+        return class extends React.Component {
+            state = {
                 config,
-            };
-        }
-
-        componentDidMount () {
-            const uaParser = Singleton.instance(UAParser);
-            const deviceData = uaParser.getDevice();
-            const browserData = uaParser.getBrowser();
-
-            this.populateData({ deviceData, browserData });
-        }
-
-        getInjections () {
-            return this.state.config
-                .filter(arg => deviceTypes.some(type => arg === type))
-                .map(device => ({
-                    [deviceToPropMap.get(device)]: this.state[deviceToPropMap.get(device)],
-                }))
-                .reduce((acc, curr) => ({ ...acc, ...curr }), {});
-        }
-
-        populateData ({ deviceData, browserData }) {
-            this.setState({
                 isChrome: evaluate(browserData, CHROME_BROWSER),
                 isIE: evaluate(browserData, IE_BROWSER),
                 isMobile: evaluate(deviceData, MOBILE_DEVICE, 'type'),
                 isFirefox: evaluate(browserData, FIREFOX_BROWSER),
-            });
-        }
+            };
 
-        render () {
-            const injections = this.getInjections();
-            return <WrappedComponent {...this.props} {...injections} />;
-        }
+            getInjections () {
+                return this.state.config
```

Comment:
Doesn't this only need to happen once? right now on every render the injections are re-calculated (filter, map, reduce...)

Reply:
Updated :)